packages/components/src/components/primitives/PinInput/types.ts

```diff
@@ -0,0 +1,20 @@
+import React, { FC } from 'react';
+import { PinInputProps as ChakraPinInputProps, PinInputFieldProps } from '@chakra-ui/react';
+
+export const pinInputVariants = ['outline', 'flushed'] as const;
+export const pinInputSizes = ['s', 'm', 'l'] as const;
+
+export type PinInputVariant = typeof pinInputVariants[number];
+export type PinInputSize = typeof pinInputSizes[number];
+
+export interface PinInputProps extends Omit<ChakraPinInputProps, 'children'> {
+  variant?: PinInputVariant;
+  size?: PinInputSize;
+  numOfDigits?: number;
+  pinInputFieldProps?: PinInputFieldProps;
+  children?: React.ReactNode;
```

Comment:
Do we really want to omit the `children` from `ChakraPinInputProps`? @TheEnmanuel23

Reply:
yes, as I mentioned [above](https://github.com/cmpsr/composer/pull/367#pullrequestreview-889119759), I omit children to change it from a required prop to an optional prop  (line 15), since we can set the fields by `numOfDigits` prop or passing directly children using composition.

By `numOfDigits`:
```jsx
 <PinInput numOfDigits={2}  ... />
```

By composition and children prop:
```jsx
 <PinInput>
    <PinInput.Field />
    <PinInput.Field />
  </PinInput>
```